File: SEP3_Client/Model/List/Message/PrivateMessageList.cs
using System.Collections.Generic;
using System.Linq;
using SEP3_Client.Model.Unit.Message;

namespace SEP3_Client.Model.List.Message
{
    public class PrivateMessageList
    {
        public List<PrivateMessage> MessageList { get; set; }

        public PrivateMessageList()
        {
            MessageList = new List<PrivateMessage>();
        }

        public void AddMessage(PrivateMessage newMessage)
        {
            MessageList.Add(newMessage);
        }

        public int GetSize()
        {
            return MessageList.Count;
        }

        public PrivateMessage GetMessageByIndex(int index)
        {
            if (index>=0&&index<MessageList.Count)
            {
                return MessageList[index];
            }
            else
            {
                return null;
            }
        }

        public PrivateMessageList GetMessageBySenderId(string id)
        {
            var messageList = new PrivateMessageList();
            foreach (var message in MessageList)
            {
                if (message.SenderId==id)
                {
                    messageList.AddMessage(message);
                }
            }
            return messageList;
        }
        
        public PrivateMessageList GetMessageByReceiverId(string id)
        {
            var messageList = new PrivateMessageList();
            foreach (var message in MessageList)
            {
                if (message.ReceiverId==id)
                {
                    messageList.AddMessage(message);
                }
            }
            return messageList;
        }
        
        public PrivateMessageList GetMessageById(string id)
        {
            var messageList = new PrivateMessageList();
            foreach (var message in MessageList)
            {
                if (message.SenderId==id||message.ReceiverId==id)
                {
                    messageList.AddMessage(message);
                }
            }
            return messageList;
        }

        public void RemoveMessageById(string senderId, string receiverId)
        {
            if (!string.IsNullOrEmpty(senderId))
            {
                if (!string.IsNullOrEmpty(receiverId))
                {
                    foreach (var message in MessageList.Where(message => message.SenderId==senderId&&message.ReceiverId==receiverId))
                    {
                        MessageList.Remove(message);
                    }
                }
                else
                {
                    foreach (var message in MessageList.Where(message => message.SenderId==senderId))
                    {
                        MessageList.Remove(message);
                    }
                }
            }
            else
            {
                if (!string.IsNullOrEmpty(receiverId))
                {
                    foreach (var message in MessageList.Where(message => message.ReceiverId==receiverId))
                    {
                        MessageList.Remove(message);
                    }
                }
            }
        }

        public PrivateMessageList Copy()
        {
            var messageList = new PrivateMessageList();
            foreach (var message in MessageList)
            {
                messageList.AddMessage(message.Copy());
            }
            return messageList;
        }
    }
}

File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/list/message/PrivateMessageList.java
package com.example.SEP3_Database.model.domain.list.message;

import com.example.SEP3_Database.model.domain.unit.message.PrivateMessage;

import java.util.ArrayList;

public class PrivateMessageList {
    private ArrayList<PrivateMessage> MessageList;

    public PrivateMessageList()
    {
        MessageList = new ArrayList<>();
    }

    public void addMessage(PrivateMessage newMessage)
    {
        if (newMessage!=null)
        {
            MessageList.add(newMessage);
        }
    }

    public int getSize()
    {
        return MessageList.size();
    }

    public PrivateMessage getMessageByIndex(int index)
    {
        if (index>=0&&index<MessageList.size())
        {
            return MessageList.get(index);
        }
        else
        {
            return null;
        }
    }

    public PrivateMessageList getMessageBySenderId(String id)
    {
        var messageList = new PrivateMessageList();
        for (int x=0;x<MessageList.size();x++)
        {
            if (MessageList.get(x).SenderId.equals(id))
            {
                messageList.addMessage(MessageList.get(x));
            }
        }
        return messageList;
    }

    public PrivateMessageList getMessageByReceiverId(String id)
    {
        var messageList = new PrivateMessageList();
        for (int x=0;x<MessageList.size();x++)
        {
            if (MessageList.get(x).getReceiverId().equals(id))
            {
                messageList.addMessage(MessageList.get(x));
            }
        }
        return messageList;
    }

    public PrivateMessageList getMessageById(String id)
    {
        var messageList = new PrivateMessageList();
        for (int x=0;x<MessageList.size();x++)
        {
            if (MessageList.get(x).SenderId.equals(id) || MessageList.get(x).getReceiverId().equals(id))
            {
                messageList.addMessage(MessageList.get(x));
            }
        }
        return messageList;
    }

    public void RemoveMessageById(String senderId, String receiverId)
    {
        ArrayList<PrivateMessage> privateMessageList = new ArrayList<>();
        if (senderId!=null)
        {
            if (receiverId!=null)
            {
                for (int x=0;x<MessageList.size();x++)
                {
                    if (!MessageList.get(x).SenderId.equals(senderId) || !MessageList.get(x).getReceiverId().equals(receiverId))
                    {
                        privateMessageList.add(MessageList.get(x));
                    }
                }
            }
            else
            {
                for (int x=0;x<MessageList.size();x++)
                {
                    if (!MessageList.get(x).SenderId.equals(senderId))
                    {
                        privateMessageList.add(MessageList.get(x));
                    }
                }
            }
        }
        else
        {
            if (receiverId!=null)
            {
                for (int x=0;x<MessageList.size();x++)
                {
                    if (!MessageList.get(x).getReceiverId().equals(receiverId))
                    {
                        privateMessageList.add(MessageList.get(x));
                    }
                }
            }
        }
    }

    public PrivateMessageList copy()
    {
        var messageList = new PrivateMessageList();
        for (int x=0;x<MessageList.size();x++)
        {
            messageList.MessageList.add(MessageList.get(x).copy());
        }
        return messageList;
    }
}
